src/wallet/connection.ts

```diff
@@ -72,8 +74,14 @@ export default class WalletServiceConnection extends BaseConnection {
 
     this.websocket = new WalletServiceWebSocket(wsOptions);
     this.websocket.on('is_online', online => this.onConnectionChange(online));
-    this.websocket.on('new-tx', payload => this.emit('new-tx', payload.data as WsTransaction));
-    this.websocket.on('update-tx', payload => this.emit('update-tx', payload.data));
+    this.websocket.on('new-tx', payload => {
+      const validatedTx = parseSchema(payload.data, wsTransactionSchema);
```

Comment:
Maybe we should put a try..catch around this or use `safeParse` directly.
Or do you think its better to have an unhandled error than a silent fail (only log an error) on new-tx?

Reply:
You are right, we shouldn't throw an error here

Fixed